lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -402,7 +429,6 @@ assignMemoryLayouts(llvm::SmallVector<std::tuple<Operation *, int, Operation *>>
                         &loadOpToIndLevelAndUse,
                     tt::ModuleAxisInfoAnalysis &axisInfoAnalysis) {
   llvm::MapVector<Operation *, LoadInfo> loadToInfo;
-
   for (auto &[op, dist, use] : loadOpToIndLevelAndUse) {
```

Comment:
I am keeping the logic for loadOpToIndLvelAndUse inside assignMemoryLayout. It is not clear to me how to split the logic between loopScheduling and SWP. This function sets fields such as
```
  ttg::SharedEncodingAttr sharedEncoding = nullptr;
  ttg::BlockedEncodingAttr blockedEncoding = nullptr;
  bool loadIsMMAV3 = false;
  bool usedByDot = false;
```
The decision of whether or not to pipeline a loadOp depends on earlier decisions based on encoding etc. Maybe we can move this whole function to loopScheduling? But we will need to duplicate some of the logic to set the fields inside SWP.
CC @pawelszczerbuk @ThomasRaoux

Reply:
This diff is not ready for detailed review. But I would appreciate looking at changes to MatmulLoopPipeline.cpp to see if the overall direction looks fine.